### PR TITLE
release/7.2.0

### PR DESCRIPTION
**Pull request by maxwellmooney13**

### CODE REVIEW

File: test/unit/media/headset.test.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            expect(sdkHeadset['sdk']).toBe(sdk);
+            expect(sdkHeadset['headsetLibrary']).toBe(headsetLibrary);
+            expect(sdkHeadset['headsetEvents$']).toStrictEqual(headsetEvents$);
+        })
+    })
+
+    describe('updateAudioInputDevice', () => {
+        it('should fetch the proper device and send it to the headset library', () => {
+            const testId = "testId";
+            const findCachedDeviceByIdAndKindSpy = jest.spyOn(sdk.media, 'findCachedDeviceByIdAndKind' as any).mockReturnValue({
```
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:38]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
            const findCachedDeviceByIdAndKindSpy = jest.spyOn(sdk.media, 'findCachedDeviceByIdAndKind' as unknown).mockReturnValue({
```

File: test/unit/media/headset.test.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            headsetLibrary.selectedImplementation.disableRetry = false;
+            headsetLibrary.selectedImplementation.isConnecting = false;
+            const showRetryResult = sdkHeadset.showRetry();
+            expect(showRetryResult).toBe(true);
+        })
+
+        it('should return false if disableRetry is false, isConnected is false but isConnecting is true', () => {
+            headsetLibrary.activeMicChange('plantronics test device');
+            headsetLibrary.selectedImplementation.isConnecting = true;
+            let showRetryResult = sdkHeadset.showRetry();
```
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:76]: Codacy has a fix for the issue: ['showRetryResult' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
            const showRetryResult = sdkHeadset.showRetry();
```

File: test/unit/client.integration.test.ts [hunk 3]
```diff
@@ -31,6 +34,11 @@ function disconnectSdk (sdk: GenesysCloudWebrtcSdk): Promise<any> {
 }
 
 describe('Client (integration)', () => {
+  Object.defineProperty(window.navigator, 'hid', { get: () => ({
+    getDevices: () => { return [] }
+  })});
+  Object.defineProperty(window.navigator, 'locks', { get: () => ({})});
+  (window as any).BroadcastChannel = BroadcastChannel;
```
Comment (davidmorton-gen) [inline test/unit/client.integration.test.ts:41]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
  (window as unknown).BroadcastChannel = BroadcastChannel;
```

File: test/unit/media/headset.test.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+let sdkHeadset: SdkHeadset;
+let headsetLibrary: HeadsetService;
+let headsetEvents$: Observable<ConsumedHeadsetEvents>;
+
+describe('SdkHeadset', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        headsetLibrary = HeadsetService.getInstance({ logger: console });
+        headsetEvents$ = headsetLibrary.headsetEvents$;
+        sdk = new SimpleMockSdk() as any;
```
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:18]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
        sdk = new SimpleMockSdk() as unknown;
```

File: test/unit/media/headset.test.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+    describe('updateAudioInputDevice', () => {
+        it('should fetch the proper device and send it to the headset library', () => {
+            const testId = "testId";
+            const findCachedDeviceByIdAndKindSpy = jest.spyOn(sdk.media, 'findCachedDeviceByIdAndKind' as any).mockReturnValue({
+                kind: 'audioinput',
+                deviceId: 'testId',
+                label: 'Test Device Mark V',
+            } as MediaDeviceInfo);
+            const activeMicChangeSpy = jest.spyOn(headsetLibrary, 'activeMicChange' as any);
```
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:43]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
            const activeMicChangeSpy = jest.spyOn(headsetLibrary, 'activeMicChange' as unknown);
```

**Review by djhouseknecht — APPROVED** [review]

File: test/unit/media/headset.test.ts (hunk 4 above)
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:18]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
        sdk = new SimpleMockSdk() as unknown;
```

File: test/unit/media/headset.test.ts [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+            headsetLibrary.activeMicChange('plantronics test device');
+            headsetLibrary.selectedImplementation.isConnecting = true;
+            const showRetryResult = sdkHeadset.showRetry();
+            expect(showRetryResult).toBe(false);
+        })
+
+        it('should return false if disableRetry is false, isConnecting is false but isConnected is true', () => {
+            headsetLibrary.activeMicChange('plantronics test device');
+            headsetLibrary.selectedImplementation.isConnected = true;
+            let showRetryResult = sdkHeadset.showRetry();
```
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:83]: Codacy has a fix for the issue: ['showRetryResult' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
            const showRetryResult = sdkHeadset.showRetry();
```

File: test/unit/client.integration.test.ts (hunk 3 above)
Comment (davidmorton-gen) [inline test/unit/client.integration.test.ts:41]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
  (window as unknown).BroadcastChannel = BroadcastChannel;
```

Comment (davidmorton-gen) [inline test/unit/client.integration.test.ts:41]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
  (window as unknown).BroadcastChannel = BroadcastChannel;
```

File: test/unit/media/headset.test.ts (hunk 4 above)
Comment (davidmorton-gen) [inline test/unit/media/headset.test.ts:18]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
        sdk = new SimpleMockSdk() as unknown;
```

File: test/unit/client.integration.test.ts (hunk 3 above)
Comment (davidmorton-gen) [inline test/unit/client.integration.test.ts:41]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9082220)
```suggestion
  (window as unknown).BroadcastChannel = BroadcastChannel;
```

**Pull request merged**